gomock/matchers.go

```diff
@@ -139,3 +190,52 @@ func Not(x interface{}) Matcher {
 func AssignableToTypeOf(x interface{}) Matcher {
 	return assignableToTypeOfMatcher{reflect.TypeOf(x)}
 }
+
+// PredFunc is a Matcher that matches if the one argument function parameter
+// returns true for the given input
+//
+// Example usage:
+//
+// 		mock.EXPECT().
+// 			Handle(gomock.PredFunc(func(x interface{}) bool { return x.(int) > 5 })).
+// 			Return(true)
+//
+func PredFunc(f func(interface{}) bool, name string) Matcher {
+	return predFuncMatcher{predFunc: f, name: name}
+}
+
+// HasPrefix is a Matcher that matches if the input has the arg as a prefix
+//
+// Example usage:
+//
+// 		mock.EXPECT().
+// 			Handle(gomock.HasPrefix("abc")).
+// 			Return(true)
+//
+func HasPrefix(prefix string) Matcher {
+	return hasPrefixMatcher{prefix: prefix}
+}
+
+// HasSuffix is a Matcher that matches if the input has the arg as a suffix
```

Comment:
nit: Can we make this commend doc a sentence with a period to match the other matchers?